src/dvc_data/fs.py

```diff
@@ -53,16 +53,18 @@ def _get_fs_path(self, path: "AnyFSPath"):
 
         entry = info["entry"]
 
-        cache_path = entry.odb.oid_to_path(value)
-
-        if entry.odb.fs.exists(cache_path):
-            return entry.odb.fs, cache_path
-
-        if not entry.remote:
-            raise FileNotFoundError
-
-        remote_fs_path = entry.remote.oid_to_path(value)
-        return entry.remote.fs, remote_fs_path
+        odb = self.index.odb_map.get(entry.key)
+        if odb:
+            cache_path = odb.oid_to_path(value)
+            if odb.fs.exists(cache_path):
+                return odb.fs, cache_path
+
+        remote = self.index.remote_map.get(entry.key)
+        if remote:
+            remote_path = remote.oid_to_path(value)
+            return remote.fs, remote_path
+
+        raise FileNotFoundError
```

Comment:
Could hide this fallback logic in a odb wrapper class that would look into cache and fallback to remote, but didn't want to get into it for now.